gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/transformation/functions/TransformGraphTransaction.java

```diff
@@ -0,0 +1,115 @@
+
+import org.apache.flink.api.common.functions.MapFunction;
+import org.gradoop.common.model.api.entities.EPGMEdgeFactory;
+import org.gradoop.common.model.api.entities.EPGMGraphHeadFactory;
+import org.gradoop.common.model.api.entities.EPGMVertexFactory;
+import org.gradoop.common.model.impl.pojo.Edge;
+import org.gradoop.common.model.impl.pojo.GraphHead;
+import org.gradoop.common.model.impl.pojo.Vertex;
+import org.gradoop.common.util.GradoopConstants;
+import org.gradoop.flink.model.api.functions.TransformationFunction;
+import org.gradoop.flink.model.impl.layouts.transactional.tuples.GraphTransaction;
+
+import java.util.stream.Collectors;
+
+/**
+ * Transformation map function for graph transactions.
+ */
+public class TransformGraphTransaction implements MapFunction<GraphTransaction, GraphTransaction> {
+
+  /**
+   * Factory to init modified graph head.
+   */
+  private final EPGMGraphHeadFactory<GraphHead> graphHeadFactory;
+  /**
+   * Graph head modification function
+   */
+  private final TransformationFunction<GraphHead> graphHeadTransFunc;
+
+  /**
+   * Factory to init modified vertex.
+   */
+  private final EPGMVertexFactory<Vertex> vertexFactory;
+
+  /**
+   * Vertex modification function
+   */
+  private final TransformationFunction<Vertex> vertexTransFunc;
+
+  /**
+   * Factory to init modified edge.
+   */
+  private final EPGMEdgeFactory<Edge> edgeFactory;
+  /**
+   * Edge modification function
+   */
+  private final TransformationFunction<Edge> edgeTransFunc;
+
+  /**
+   * Constructor
+   *
+   * @param graphHeadFactory graph head factory
+   * @param graphHeadTransFunc graph head transformation function
+   * @param vertexFactory vertex factory
+   * @param vertexTransFunc vertex transformation function
+   * @param edgeFactory edge factory
+   * @param edgeTransFunc edge transformation function
+   */
+  public TransformGraphTransaction(EPGMGraphHeadFactory<GraphHead> graphHeadFactory,
+    TransformationFunction<GraphHead> graphHeadTransFunc, EPGMVertexFactory<Vertex> vertexFactory,
+    TransformationFunction<Vertex> vertexTransFunc, EPGMEdgeFactory<Edge> edgeFactory,
+    TransformationFunction<Edge> edgeTransFunc) {
+    this.graphHeadFactory = graphHeadFactory;
+    this.graphHeadTransFunc = graphHeadTransFunc;
+    this.vertexFactory = vertexFactory;
+    this.vertexTransFunc = vertexTransFunc;
+    this.edgeFactory = edgeFactory;
+    this.edgeTransFunc = edgeTransFunc;
+  }
+
+  @Override
+  public GraphTransaction map(GraphTransaction transaction) throws Exception {
+    if (graphHeadTransFunc != null) {
+      transaction.setGraphHead(graphHeadTransFunc.apply(
+        transaction.getGraphHead(),
+        graphHeadFactory.initGraphHead(
+          transaction.getGraphHead().getId(), GradoopConstants.DEFAULT_GRAPH_LABEL)));
+    }
+
+    if (vertexTransFunc != null) {
+      transaction.setVertices(transaction.getVertices().stream()
+        .map(vertex -> vertexTransFunc.apply(
```

Comment:
Do we really want to use lambda-expressions? According to the [Flink documentation](https://ci.apache.org/projects/flink/flink-docs-release-1.3/dev/java8.html) these are only supported under very specific circumstances.